src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1090,12 +1090,6 @@ void CodeGen::genHWIntrinsic_R_R_RM_I(
 
     assert(targetReg != REG_NA);
 
-    if (op2->isContained() || op2->isUsedFromSpillTemp())
-    {
-        assert(HWIntrinsicInfo::SupportsContainment(node->GetHWIntrinsicId()));
-        assertIsContainableHWIntrinsicOp(compiler->m_pLowering, node, op2);
-    }
-
     if (ins == INS_insertps)
     {
         if (op1Reg == REG_NA)
```

Comment:
Could you also fix this line to check `op1->isContained()`, just to make it more explicit.

We have some contexts where register assignments are left in some nodes even if `isUsedFromReg` is false and it should help ensure we aren't doing the incorrect thing: https://github.com/dotnet/runtime/issues/105837